### PR TITLE
refactor: Extract ItemStackRenderer.renderInWorld into new class

### PR DESCRIPTION
**Pull request by szuend**
The ItemStackRenderer is very stateful and is used in various places for different reasons. This makes it hard to follow what is going on.

This PR extracts the hot 'renderInWorld' function into a separate class. It's invoked for every traveling item to render the item itself in the transport box. The only important bit is that we need an 'EntityItem' cache for performance reasons.

Why can the PR remove the custom 'RenderItem' subclass? Because it was never actually used: 'renderInWorld' invokes a render function on the "RenderManager.instance", which in turn invokes it's own "RenderItem" class, not the custom one. The custom one was never used for "renderInWorld".

Drive-by: We can create an "RenderItem" of our own and invoke it's 'doRender' directly instead of going via the RenderManager, who has to do a lookup in a map.

### CODE REVIEW

File: src/main/java/logisticspipes/utils/item/ItemStackRenderer.java [hunk 1]
```diff
@@ -260,39 +256,6 @@ public void renderInGui() {
...
-        assert renderManager != null;
-        assert renderItem != null;
-
-        ItemIdentifier itemId = itemIdentifierStack.getItem();
-        EntityItem entityItem = entityCache.get(itemId);
-        if (entityItem == null) {
-            entityItem = new EntityItem(worldObj, 0, 0, 0, itemId.makeNormalStack(1));
-            entityItem.hoverStart = 0;
-            entityCache.put(itemId, entityItem);
-        }
```
Comment (OneEyeMaker) [inline src/main/java/logisticspipes/utils/item/ItemStackRenderer.java:273]: Is `entityCache` field still required and used in this class?

Comment (szuend) [reply src/main/java/logisticspipes/utils/item/ItemStackRenderer.java:273]: Nice catch! I was planning to remove some other unused things from `ItemStackRenderer` but forgot that removing `entityCache` should be part of this PR.

**Review by OneEyeMaker — APPROVED** [review]
